kong/router.lua

```diff
@@ -522,7 +522,8 @@ do
 
     [MATCH_RULES.URI] = function(route_t, ctx)
       do
-        local uri_t = route_t.uris[ctx.hits.uri or ctx.req_uri]
+        local uri_t = route_t.uris[ctx.req_uri] or
+                      route_t.uris[ctx.hits.uri]
```

Comment:
Ditto: while the impact is lesser, we still disregard the hint (which serves no purpose anymore) in favor of the request path.